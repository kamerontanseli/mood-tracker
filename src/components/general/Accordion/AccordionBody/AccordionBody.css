.AccordionBody {
  margin-top: 20px;
}