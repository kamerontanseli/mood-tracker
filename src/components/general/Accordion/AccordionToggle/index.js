import React from "react";
import Button from '../../Button'

const AccordionToggle = ({ toggled, ...props }) => (
  <Button {...props}>
    {toggled ? <span>&uarr;</span> : <span>&darr;</span>}
  </Button>
);

export default AccordionToggle;
