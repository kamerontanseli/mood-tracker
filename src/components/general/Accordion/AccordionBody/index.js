import "./AccordionBody.css";
import React from "react";

const AccordionBody = ({ children }) => (
  <div className="AccordionBody">{children}</div>
);

export default AccordionBody;
