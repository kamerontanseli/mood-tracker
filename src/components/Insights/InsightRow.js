import React from "react";
import Tag from "../general/Tag";
import Accordion from "../general/Accordion";
import AccordionHead from "../general/Accordion/AccordionHead";
import AccordionHeadItem from "../general/Accordion/AccordionHeadItem";
import AccordionToggle from "../general/Accordion/AccordionToggle";
import AccordionBody from "../general/Accordion/AccordionBody";
import { getMonth } from "../../utils/date";
import { pad, getMood } from "../../utils/string";

const InsightRow = ({ insight }) => {
  const date = new Date(insight.date);

  return (
    <Accordion>
      {({ onToggle, toggled }) => (
        <React.Fragment>
          <AccordionHead>
            <AccordionHeadItem>
              {getMonth(date.getMonth())} {date.getDate()}
            </AccordionHeadItem>
            <AccordionHeadItem>
              {pad(date.getHours())}:{pad(date.getMinutes())}
            </AccordionHeadItem>
            <AccordionHeadItem>{getMood(insight.mood)}</AccordionHeadItem>
            <AccordionHeadItem>
              <AccordionToggle onClick={onToggle} toggled={toggled} />
            </AccordionHeadItem>
          </AccordionHead>
          {toggled && (
            <AccordionBody>
              <div>
                {insight.feelings.map((feeling, index) => (
                  <Tag key={index}>{feeling}</Tag>
                ))}
              </div>
              <p>{insight.comment}</p>
            </AccordionBody>
          )}
        </React.Fragment>
      )}
    </Accordion>
  );
};

export default InsightRow;
