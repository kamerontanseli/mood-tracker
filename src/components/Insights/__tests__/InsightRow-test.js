import React from "react";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import InsightRow from "../InsightRow";
import AccordionToggle from "../../general/Accordion/AccordionToggle";
import AccordionBody from "../../general/Accordion/AccordionBody";

describe("render", () => {
  it("should render", () => {
    const snapshot = renderer.create(
      <InsightRow
        insight={{
          mood: 5,
          feelings: ["optimistic", "happy"],
          comment: "wdqwdwdqdw",
          date: 1540507773565
        }}
      />
    );
    expect(snapshot).toMatchSnapshot();
  });
  it("should show AccordionBody when AccordionToggle is clicked", () => {
    const wrapper = mount(
      <InsightRow
        insight={{
          mood: 5,
          feelings: ["optimistic", "happy"],
          comment: "wdqwdwdqdw",
          date: 1540507773565
        }}
      />
    );
    wrapper.find(AccordionToggle).simulate('click')
    wrapper.update()
    expect(wrapper.find(AccordionBody).exists()).toBe(true)
  });
});
