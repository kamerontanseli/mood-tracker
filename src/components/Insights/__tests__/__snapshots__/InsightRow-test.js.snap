// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render 1`] = `
<div
  className="Panel"
>
  <div
    className="AccordionHead"
  >
    <div
      className="AccordionHeadItem"
    >
      Oct
       
      25
    </div>
    <div
      className="AccordionHeadItem"
    >
      23
      :
      49
    </div>
    <div
      className="AccordionHeadItem"
    >
      Good
    </div>
    <div
      className="AccordionHeadItem"
    >
      <button
        className="Button"
        onClick={[Function]}
      >
        <span>
          â†“
        </span>
      </button>
    </div>
  </div>
</div>
`;
